src/GraphPanel.java

```diff
@@ -3,20 +3,19 @@
 import java.awt.event.*;
 import java.awt.geom.GeneralPath;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class GraphPanel extends JPanel {
-    private double zoom = 1.0;
+    private final List<ParametricFunction> polynomials = new ArrayList<>();
     private Vector2D offset = new Vector2D(0,0);
-    private double scale = 50.0;
-    private final Color backgroundColor = new Color(0.13f, 0.16f, 0.2f);
     private Vector2D lastMousePosition = null;
+    private double zoom = 1.0;
```

Comment:
no need to have both zoom and scale, as they are directly proportional to one another and can be easily calculated (maybe implement a separate function, something like "getZoom" or "getScale" which does this on the fly). In general, we want as little state as necessary.